src/lib/y2network/network_manager/connection_config_writers/hsi.rb

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/network_manager/connection_config_writers/qeth"
+
+module Y2Network
+  module NetworkManager
+    module ConnectionConfigWriters
+      # This class is responsible for writing the information from a ConnectionConfig::Hsi
+      # object to the underlying system.
+      class Hsi < Qeth
```

Comment:
An empty class is something that certainly looks weird.

Reply:
Seems to be a result of this, which couples the name of the file to its format, which I'm not sure is 100% correct.
https://github.com/yast/yast-network/blob/166c75353147a0c3a58612113a8fa8edca3ac2be/src/lib/y2network/network_manager/connection_config_writer.rb#L70

Reply:
We could move this specific options to a **mixin** to be included by **Qeth** and **Hsi** classes which are written in the same way. But that fix nothing and we already have the same in wicked writers. So, for a bug report looks weird to change the current approach.

Reply:
The same approach were already used by several wicked writers which does not different in options from an **ethernet** connection like:

- https://github.com/yast/yast-network/blob/166c75353147a0c3a58612113a8fa8edca3ac2be/src/lib/y2network/wicked/connection_config_writers/hsi.rb
- https://github.com/yast/yast-network/blob/166c75353147a0c3a58612113a8fa8edca3ac2be/src/lib/y2network/wicked/connection_config_writers/qeth.rb
- https://github.com/yast/yast-network/blob/166c75353147a0c3a58612113a8fa8edca3ac2be/src/lib/y2network/wicked/connection_config_writers/lcs.rb

Reply:
Modifying the way the connection writers are selected is probably too much for a bug report, so it is something completely discarded by now.